### PR TITLE
[windows][filesystem] Preserve temporary file suffixes

### PR DESCRIPTION
**Pull request by garbear**
## Description

AI use:

Preserve the requested suffix in the Windows implementation of `temp_file_path()`.

Previously, the suffix was ignored and temporary files retained the `.tmp` extension generated by `GetTempFileNameW()`. Requests for `.m3u` playlists or `.chd` media therefore produced filenames that failed format detection.

Append the requested suffix to the generated path, matching the suffix preservation already provided by the POSIX implementation. Empty-suffix behavior remains unchanged.

Add a Windows regression test covering empty, `.chd`, `.m3u`, and `.sav` suffixes, including file creation, writing, and cleanup.

## Motivation and context

https://github.com/xbmc/xbmc/pull/29210 revealed this existing Windows bug. Its disc-persistence tests depend on temporary files retaining their requested extensions so playlists and supported media are recognized.

This fix is needed for those tests to pass on Windows. It resolves the 22 affected failures without changing disc-handling logic or weakening test assertions.

## How has this been tested?

Tested on Windows x64 using the Release configuration with Visual Studio 2026 and CMake.

- Reproduced all 22 failures on the `persist-discs` branch before applying the fix: 58 passed and 22 failed across the two affected disc suites.
- Confirmed the new suffix regression fails against the original Windows implementation for `.chd`, `.m3u`, and `.sav`.
- After applying the fix, all 81 tests in the affected disc suites and the new filesystem regression passed.
- Ran the full Windows suite on `persist-discs`: **4,239 passed, 0 failed**, with 753 existing disabled tests.

## Types of change

- [x] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement (v22 new feature polish)** (non-breaking change which improves a new feature in v22)
- [ ] **Improvement (other)** (non-breaking change which improves other existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **Student submission** (PR was done for educational purposes and will be reviewed with support)
- [ ] **None of the above** (please explain below)

### CODE REVIEW

File: xbmc/platform/win32/Filesystem.cpp [hunk 1]
```diff
@@ -153,7 +153,7 @@ std::string temp_file_path(const std::string&, std::error_code& ec)
   }
 
   ec.clear();
-  return win::FromW(lpTempPathBuffer);
+  return win::FromW(lpTempPathBuffer) + suffix;
```
[the remark this answers is not in the document]
Comment (garbear) [reply xbmc/platform/win32/Filesystem.cpp]: Fixed

**Comment by garbear** [comment]
Applied codex suggestion. @codex review

**Comment by garbear** [comment]
Unrelated OSX-ARM64 failure (fixed by https://github.com/xbmc/xbmc/pull/29231). Related UWP failure, will push a fix.
